### DISCUSSION TITLE
Analog comparator with timer function

### DISCUSSION THREAD

**Discussion by codev123** [stm32duino/Arduino_Core_STM32#2246] [Q&A] [answered]
Dear sir, thank you for stm32 Arduino core. I have an issue, i couldn't not find any analog comparator based help on this Arduino code. Can you please help/guide me on how to add analog comparator support with hardware timer to watch the comparator output interrupt and find duration in using timer. 
Thankyou sir

**Comment by fpistm** [comment]
Hi @codev123 
I move this topic from issue to discussion. This is not an issue within the core.
You can also go to the [forum](https://www.stm32duino.com) to get support from the community.
Could you be more explicit on your request, you said 

> i couldn't not find any analog comparator based help on this Arduino code.

Does it mean you saw this kind of feature on other core? If true, then share it.
Your request seems to request code snippet?

**Reply by codev123** [reply]
I have seen comparator feature on Arduino core, avr atmega328p microcontroller's core. Sorry for confusion. I wanted to know how can i implement comparator based code with timer to find out duration of state change on comparator output. If there is any code snippet available it will be big help.
Thankyou sir

**Reply by fpistm** [reply]
I guess you want to use the COMP feature of the STM32.
Currently, I have no code for this, you can try to enable it by defining the `HAL_COMP_MODULE_ENABLED`.
https://github.com/stm32duino/Arduino_Core_STM32/wiki/HAL-configuration#hal-modules-configuration
Then implement yourself its usage.

**Reply by fpistm** [reply, marked as answer]
Here a user succeed to use it:
https://www.stm32duino.com/viewtopic.php?t=2060
